lib/apiScenario/postmanCollectionRunnerClient.ts

```diff
@@ -716,27 +764,68 @@ pm.test("Stopped TestProxy recording", function() {
     if (postScripts.length > 0) {
       PostmanHelper.addEvent(item.events, "test", postScripts);
     }
+    generatePostmanAssertion({ step, type: "stepCall", item, opts: this.opts });
   }
 
   private lroPoll(
     itemGroup: ItemGroup<Item>,
     item: Item,
+    step: StepRestCall | StepArmTemplate,
     baseUri: string,
-    postScripts: string[],
-    checkStatus: boolean = false,
-    responseAssertion?: StepResponseAssertion
+    postScripts: string[]
   ) {
+    const finalStateVia =
+      step.type === "restCall"
+        ? step?.operation?.[xmsLongRunningOperationOptions]?.[xmsLongRunningOperationOptionsField]
+        : undefined;
+    const isManagementPlane = step.type === "armTemplateDeployment" || step.isManagementPlane;
     if (this.opts.skipLroPoll) return;
+    const url = item.request.url;
+    const urlStr = `${baseUri}${url.getPathWithQuery()}`;
 
+    // For ARM put or patch operations , by default the final get is original url.
+    const isArmResourceCreate =
+      isManagementPlane && ["put", "patch"].includes(item.request.method.toLowerCase());
     postScripts.push(
       `
-const pollingUrl = pm.response.headers.get("Location") || pm.response.headers.get("Azure-AsyncOperation") || pm.response.headers.get("Operation-Location");
+ // RPC-Async-V1-06 x-ms-long-running-operation-options should indicate the type of response header to track the async operation.
+ function getLroFinalGetUrl(finalStateVia) {
+  if (!finalStateVia) {
+    // by default, the final url header is Location for ARM, Operation-Location for dataplane.
+    return ${
+      isArmResourceCreate
+        ? `'${urlStr}'`
+        : `pm.response.headers.get("Location") || pm.response.headers.get("Operation-Location")`
+    }
+  }
+  switch (finalStateVia) {
+    case "location": {
+      return pm.response.headers.get("Location");
```

Comment:
do we assert the existence of the expected header?

Reply:
Yes, in the pre check of the final get